### PR TITLE
Kubernetes info and health clean-up + tests

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicator.java [hunk 1]
```diff
@@ -16,43 +16,44 @@
...
 import io.kubernetes.client.openapi.models.V1Pod;
 
 import org.springframework.cloud.kubernetes.commons.AbstractKubernetesHealthIndicator;
 import org.springframework.cloud.kubernetes.commons.PodUtils;
+import org.springframework.util.CollectionUtils;
 
 /**
  * @author Ryan Baxter
  */
 public class KubernetesClientHealthIndicator extends AbstractKubernetesHealthIndicator {
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicator.java:32]: minor things, `Collections::singletonMap`, `final`, move things "logically" together for easier read of the code

File: spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientInfoContributor.java [hunk 2]
```diff
@@ -38,18 +39,18 @@ public KubernetesClientInfoContributor(PodUtils<V1Pod> utils) {
 	@Override
 	public Map<String, Object> getDetails() {
 		V1Pod current = this.utils.currentPod().get();
-		Map<String, Object> details = new HashMap<>();
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientInfoContributor.java:41]: same reasonings as above

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorInsideTests.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * @author wind57
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = App.class)
+public class KubernetesClientHealthIndicatorInsideTests {
+
+	@Autowired
+	private ApplicationContext context;
+
+	// test that the bean responsible for health is present.
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorInsideTests.java:37]: the comment should make it clear what this tests

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorNotInsideTests.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * @author wind57
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = App.class,
+		properties = { "management.health.kubernetes.enabled=false" })
+public class KubernetesClientHealthIndicatorNotInsideTests {
+
+	@Autowired
+	private ApplicationContext context;
+
+	// test that the bean responsible for info contribution is NOT present.
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorNotInsideTests.java:39]: again, the comment makes it clear why this exists

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorTests.java [hunk 5]
```diff
@@ -43,48 +54,32 @@
...
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.INSIDE)).isEqualTo(false);
+		Map<String, Object> details = healthIndicator.getDetails();
+
+		assertThat(details.containsKey(INSIDE)).isTrue();
+		assertThat(details.get(INSIDE)).isEqualTo(false);
 	}
 
 	@Test
-	void getDetailsInsideTest() throws Exception {
-		Map<String, String> labels = new HashMap<>();
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorTests.java:55]: I've moved the stubbing of the pod, as well as the field names it contained to `StubProvider`. This is used in both health and info tests.

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorTests.java [hunk 6]
```diff
@@ -43,48 +54,32 @@
...
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.HOST_IP))
-				.isEqualTo("123.456.789.1");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.POD_IP)).isEqualTo("127.0.0.1");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.NODE_NAME)).isEqualTo("nodeName");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.SERVICE_ACCOUNT))
-				.isEqualTo("serviceAccount");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.POD_NAME)).isEqualTo("mypod");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.NAMESPACE)).isEqualTo("default");
-		assertThat(healthIndicator.getDetails().get(KubernetesClientHealthIndicator.LABELS)).isEqualTo(labels);
+		assertThat(details.get(HOST_IP)).isEqualTo(STUB_HOST_IP);
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/KubernetesClientHealthIndicatorTests.java:76]: I find this a lot easier to read and understand what is being tested

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/profile/KubernetesClientProfileEnvironmentPostProcessorNoProfileTests.java [hunk 7]
```diff
@@ -32,27 +29,15 @@
 /**
  * @author Ryan Baxter
  */
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/profile/KubernetesClientProfileEnvironmentPostProcessorNoProfileTests.java:31]: just some minor clean-up

File: spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/profile/KubernetesClientProfileEnvironmentPostProcessorTests.java [hunk 8]
```diff
@@ -34,27 +31,15 @@
 /**
  * @author Thomas Vitale
  */
-@SpringBootTest(properties = { ENV_SERVICE_HOST + "=10.0.0.1", ENV_SERVICE_PORT + "=80" },
-		classes = { KubernetesClientProfileEnvironmentPostProcessorTests.App.class })
+@SpringBootTest(properties = { ENV_SERVICE_HOST + "=10.0.0.1", ENV_SERVICE_PORT + "=80" }, classes = { App.class })
```
Comment (wind57) [inline spring-cloud-kubernetes-client-autoconfig/src/test/java/org/springframework/cloud/kubernetes/client/profile/KubernetesClientProfileEnvironmentPostProcessorTests.java:34]: minor clean-up

**Review by ryanjbaxter — APPROVED** [review]

**Pull request merged**